src/new_index/mempool.rs

```diff
@@ -285,6 +285,48 @@ impl Mempool {
         &self.backlog_stats.0
     }
 
+    pub fn old_txids(&self) -> HashSet<Txid> {
+        return HashSet::from_iter(self.txstore.keys().cloned());
+    }
+
+    pub fn download_new_mempool_txs(daemon: &Daemon, old_txids: &HashSet<Txid>, new_txids: &HashSet<Txid>) -> Vec<Transaction> {
+        let t = Instant::now();
+
+        let txids: Vec<&Txid> = (*new_txids).difference(old_txids).collect();
+        let tranactions = match daemon.gettransactions(&txids) {
+            Ok(txs) => txs,
+            Err(err) => {
+                warn!("failed to get {} transactions: {}", txids.len(), err); // e.g. new block or RBF
+                vec![] // return an empty vector if there's an error
+            }
+        };
+
+        log_fn_duration("mempool::download_new_mempool_txs", t.elapsed().as_micros());
+        return tranactions;
+    }
+
+    pub fn update_quick(&mut self, to_add: &Vec<Transaction>, to_remove: &HashSet<&Txid>) -> Result<()> {
+        let t = Instant::now();
+        let _timer = self.latency.with_label_values(&["update"]).start_timer();
+
+        // Add new transactions
+        self.add(to_add.clone());
+        // Remove missing transactions
+        self.remove(to_remove.clone());
+
+        self.count
+            .with_label_values(&["txs"])
+            .set(self.txstore.len() as f64);
+
+        // Update cached backlog stats (if expired)
+        if self.backlog_stats.1.elapsed() > Duration::from_secs(BACKLOG_STATS_TTL) {
+            self.update_backlog_stats();
+        }
+
+        log_fn_duration("mempool::update_quick", t.elapsed().as_micros());
+        Ok(())
+    }
+
     pub fn update(&mut self, daemon: &Daemon) -> Result<()> {
```

Comment:
The old `update()` code should be removed.

It can be replaced with a static `Mempool::update(RwLock<Mempool>, &Daemon)`, as suggested in a previous comment. The `update_quick()` code could be merged into that too.

Reply:
Just a heads up -- I had to remove this part of the code, since we no longer have access to `self` in the static method. Is that okay?

```
        let _timer = self.latency.with_label_values(&["update"]).start_timer();
```

Reply:
Not the prettiest, but it could obtain a read lock and get the timer out of it, e.g. `let _timer = mempool.read().unwrap().latency.with_label_values(&["update"]).start_timer();`

If you want to measure inner-function durations (e.g. just for downloading the txs), you could also clone `latency` to have a copy of it that can be used multiple times.